src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -34,6 +34,54 @@ void FilterRemovedNodes(std::shared_ptr<gcs::GcsClient> gcs_client,
   }
 }
 
+/// Update object location data based on response from the owning core worker.
+bool UpdateObjectLocations(const rpc::GetObjectLocationsOwnerReply &location_reply,
+                           const Status &status, const ObjectID &object_id,
+                           std::shared_ptr<gcs::GcsClient> gcs_client,
+                           std::unordered_set<NodeID> *node_ids, std::string *spilled_url,
+                           NodeID *spilled_node_id, size_t *object_size) {
+  bool is_updated = false;
+
+  std::unordered_set<NodeID> new_node_ids;
+
+  if (!status.ok()) {
+    RAY_LOG(INFO) << "Failed to return location updates to subscribers  for " << object_id
+                  << ": " << status.ToString()
+                  << ", assuming that the object was freed or evicted.";
+    *node_ids = new_node_ids;
+    is_updated = true;
```

Comment:
Isn't it supposed to be false?

Reply:
No, we want to update. We consider a failed fetch of locations to indicate that the object was freed or evicted, so we'd want to invoke all subscriber callbacks with empty locations.

Reply:
Makes sense. Can you write a comment here?

Reply:
Sure!